PurchasesCoreSwift/Attribution/AttributionFetcher.swift

```diff
@@ -0,0 +1,139 @@
+//
+// Created by Andrés Boedo on 4/8/21.
+// Copyright (c) 2021 Purchases. All rights reserved.
+//
+
+import Foundation
+#if os(iOS) || os(tvOS)
+import UIKit
+#elseif os(watchOS)
+import WatchKit
+#endif
+
+enum AttributionFetcherError: Error {
+
+    case identifierForAdvertiserUnavailableForPlatform
+    case identifierForAdvertiserFrameworksUnavailable
+
+}
+
+@objc(RCAttributionFetcher) public class AttributionFetcher: NSObject {
+    private let attributionFactory: AttributionTypeFactory
+    private let systemInfo: SystemInfo
+
+    #if os(watchOS) || os(macOS) || targetEnvironment(macCatalyst)
+    private let appTrackingTransparencyRequired = false
+    #else
+    private let appTrackingTransparencyRequired = true
+    #endif
```

Comment:
xcode's formatter actually moves these to the start of the line, i.e.: 
```swift
#if os(watchOS) || os(macOS) || targetEnvironment(macCatalyst)
     private let appTrackingTransparencyRequired = false
#else
     private let appTrackingTransparencyRequired = true
#endif
```
Thoughts on what's better? I kinda like having them more integrated into the code, but I like being able to rely on the auto-formatter.

Reply:
I like it without indentation, but also agree- we shouldn't fight the auto-formatter. So I would go with whatever the formatter does from now on.